java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -89,6 +92,15 @@ Map<Endpoint, Pattern> getCompiledAllowedEndpoints() {
         return compiledAllowedEndpoints;
     }
 
+    SchemaRuleUtils.JsonSchemaFilter getRootDefinitions() {
+        if (rootDefinitions == null) {
+            synchronized ($writeLock) {
+                rootDefinitions = SchemaRuleUtils.JsonSchemaFilter.builder().definitions(rules.getDefinitions()).build();
+            }
```

Comment:
for doing the double locking singleton properly you need to check again in the lock

```java
synchronized ($writeLock) {
  if (rootDefinitions == null) {
     rootDefinitions = SchemaRuleUtils.JsonSchemaFilter.builder().definitions(rules.getDefinitions()).build();
  }
}
```